packages/block-editor/src/components/colors-gradients/style.scss

```diff
@@ -120,3 +121,29 @@ $swatch-gap: 12px;
 		flex-shrink: 0;
 	}
 }
+
+.block-editor-panel-color-gradient-settings__reset {
+	position: absolute;
+	right: 0;
+	top: 8px;
+	margin: auto $grid-unit auto;
+	opacity: 0;
+	transition: opacity 0.1s ease-in-out;
+	@include reduce-motion("transition");
+
+	// Override the button style.
+	&.components-button.has-icon {
+		border-radius: 0;
+		height: $button-size-small;
+		width: $button-size-small;
+		min-width: 0;
+		padding: 0;
+
+	}
```

Comment:
If we use the size prop of the `Button` component, we should be able to avoid using `.components-`-prefixed CSS selectors:

```diff
diff --git a/packages/block-editor/src/components/colors-gradients/style.scss b/packages/block-editor/src/components/colors-gradients/style.scss
index beddda7922..3c25d6916d 100644
--- a/packages/block-editor/src/components/colors-gradients/style.scss
+++ b/packages/block-editor/src/components/colors-gradients/style.scss
@@ -131,14 +131,8 @@ $swatch-gap: 12px;
        transition: opacity 0.1s ease-in-out;
        @include reduce-motion("transition");
 
-       // Override the button style.
-       &.components-button.has-icon {
-               border-radius: 0;
-               height: $button-size-small;
-               width: $button-size-small;
-               min-width: 0;
-               padding: 0;
-
+       &.block-editor-panel-color-gradient-settings__reset {
+               border-radius: $radius-small;
        }
 
        .block-editor-panel-color-gradient-settings__dropdown:hover + &,
diff --git a/packages/block-editor/src/components/global-styles/color-panel.js b/packages/block-editor/src/components/global-styles/color-panel.js
index 6104d25349..43b0b5899e 100644
--- a/packages/block-editor/src/components/global-styles/color-panel.js
+++ b/packages/block-editor/src/components/global-styles/color-panel.js
@@ -247,6 +247,7 @@ function ColorPanelDropdown( {
                                                                        label={ __( 'Reset' ) }
                                                                        aria-label={ __( 'Reset' ) }
                                                                        className="block-editor-panel-color-gradient-settings__reset"
+                                                                       size="small"
                                                                        icon={ resetIcon }
                                                                        onClick={ () => {
                                                                                resetValue();
```

We might also want to remove the `border-radius` , but without it all the corners won't be applied correctly:

![border-radius](https://github.com/user-attachments/assets/5dbfe1aa-319d-47d2-a8ee-0345e26ce903)

Ideally, we should adjust our CSS selectors so they don't conflict, but for no,w I think it's ok to apply an explicit border-radius.